### PR TITLE
Add implementation of Npm module

### PR DESCRIPTION
**Pull request by agc93**
So I don't know where your work was up to @gep13 but I just threw this together this afternoon with support for most of the stuff I think people would want to use.

Before we go publishing/publicizing this, probably worth adding something to the docs on the differences between this and Cake.Npm to avoid confusion.

Resolves #1

### CODE REVIEW

**Comment by agc93** [comment]
@gep13 Given discussions yesterday, I'm happy to take this into my own repo until one of us gets a chance to Wyam- and Recipe-ify this? Can publish as-is (once #3 is resolved), then migrate to the org once one of us has the time to do the migration, perhaps?

Just a thought :smile:

**Comment by gep13** [comment]
@agc93 I am not sure that I see any real benefit in moving it out of the repo, only to pull it back in again at some point in the future.  As I mentioned, usage of Cake.Recipe and Wyam is not mandatory in order for a repository to live in the Cake Contrib Organisation.  I would go as far as to say it is a strong preference, but not a blocking requirement.

I think this module is fine where it is, and we can work on bringing it into line with the other repositories in the Organisation once Cake.Recipe is capable to support it.

File: src/Cake.Npm.Module/Library.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (gep13) [inline src/Cake.Npm.Module/Library.cs]: What is this file for?

Comment (agc93) [inline src/Cake.Npm.Module/Library.cs]: Oh woops. That's just the `dotnet new` template file I forgot to remove. Will remove and push.

Comment (agc93) [inline src/Cake.Npm.Module/Library.cs]: @gep13 Fixed in 1193d7a

**Pull request merged**